### PR TITLE
#508 - Added CLI ref and Debugging conceptual content for metrics

### PR DESCRIPTION
**Pull request by donntrenton**
## PR description

Per #508, added CLI ref and Debugging conceptual content for metrics using Prometheus

## Fixed Issue(s)
#508

### CODE REVIEW

File: docs/Using-Pantheon/Debugging.md [hunk 1]
```diff
@@ -3,10 +3,52 @@ description: Frequently asked questions FAQ and answers for troubleshooting Pant
...
+
+* Navigate to the directory in which you installed the Pantheon binary and run the command with the metrics options;
+for example:
+
+    ```bash
+    bin/pantheon --dev-mode --network-id="2018" --miner-enabled --miner-coinbase fe3b557e8fb62b89f4916b721be55ceb828dbd73
+    --rpc-cors-origins "all" --ws-enabled --rpc-enabled --metrics-enabled
+    ```
+
+* In another terminal, run Prometheus.
```
Comment (shemnon) [inline docs/Using-Pantheon/Debugging.md]: Prometheus needs to be configured to poll pantheon.  For example this yaml fragment can be added to the scrape_configs block of the prometheus.yml file

```- job_name: pantheon-dev
  scrape_interval: 15s
  scrape_timeout: 10s
  metrics_path: /metrics
  scheme: http
  static_configs:
  - targets:
    - 127.0.0.1:9545```

File: docs/Reference/Pantheon-CLI-Syntax.md [hunk 2]
```diff
@@ -223,6 +223,25 @@ max-trailing-peers=2
...
+
+### metrics-listen
+
+```bash tab="Syntax"
+--metrics-listen
+```
+
+Specifies the host and port on which the [Prometheus](https://prometheus.io/) monitoring service accesses Pantheon
+client metrics. The default is `127.0.0.1:9545`. The metrics server respects the [`--host-whitelist`](#host-whitelist)
+flag for DNS rebind attack protection on RPC.
```
Comment (MadelineMurray) [inline docs/Reference/Pantheon-CLI-Syntax.md]: Is this point about the metrics server from the implementation point of view?  Does the user need to specify https://prometheus.io/ using the `--host-whitelist` option?  If so, I think that's the information we need to specify here.

File: docs/Reference/Pantheon-CLI-Syntax.md [hunk 3]
```diff
@@ -223,6 +223,25 @@ max-trailing-peers=2
 Specifies the maximum P2P peer connections for peers that are trailing behind the local chain head. 
 The default is unlimited but the number of trailing peers cannot exceed the value specified by [`--max-peers`](#max-peers).
 
+### metrics-enabled
+
+```bash tab="Syntax"
+--metrics-enabled
+```
+
+Set to `true` to enable the [Prometheus](https://prometheus.io/) monitoring service to access Pantheon client metrics.
```
Comment (MadelineMurray) [inline docs/Reference/Pantheon-CLI-Syntax.md]: Suggest changing 'Pantheon client' to 'Pantheon' to be consistent with rest of documentation

File: docs/Using-Pantheon/Debugging.md [hunk 4]
```diff
@@ -3,10 +3,52 @@ description: Frequently asked questions FAQ and answers for troubleshooting Pant
...
\ No newline at end of file
+You can monitor node performance using the [`debug_metrics`](../Reference/JSON-RPC-API-Methods.md#debug_metrics)
+JSON-RPC API method.
+
+## Monitor Node Performance Using Third-Party Clients
+
+You can monitor node performance using the [Prometheus](https://prometheus.io/) monitoring and alerting service.
+You can also visualize the collected data using [Grafana](https://grafana.com/).
+
+Use the Pantheon CLI option [`--metrics-enabled`](../Reference/Pantheon-CLI-Syntax.md#metrics-enabled) to allow
```
Comment (MadelineMurray) [inline docs/Using-Pantheon/Debugging.md]: Suggest rewording as 
Set to the `--metrics-enabled` command line option to `true` to enable the [Prometheus](https://prometheus.io/) monitoring service access to Pantheon metrics.

- Given this is Pantheon documentation I think we can assume we're talking about Pantheon command line options.
- I'd be explicit about how to set the option here so they don't have to go from here to the CLI reference just to find out it needs set to true.

File: docs/Reference/Pantheon-CLI-Syntax.md (hunk 3 above)
Comment (MadelineMurray) [reply docs/Reference/Pantheon-CLI-Syntax.md]: I think we need a link to the conceptual debugging content as well - otherwise if they come here first, they won't know it exists.

File: docs/Using-Pantheon/Debugging.md [hunk 5]
```diff
@@ -3,10 +3,52 @@ description: Frequently asked questions FAQ and answers for troubleshooting Pant
...
+Prometheus to access Pantheon client metrics.
```
Comment (MadelineMurray) [inline docs/Using-Pantheon/Debugging.md]: 'Pantheon client' -> 'Pantheon'

File: docs/Using-Pantheon/Debugging.md [hunk 6]
```diff
@@ -3,10 +3,52 @@ description: Frequently asked questions FAQ and answers for troubleshooting Pant
...
+
+While Prometheus is running on your system, it can consume the Pantheon client data directly for monitoring.
```
Comment (MadelineMurray) [inline docs/Using-Pantheon/Debugging.md]: While Prometheus is running on your system, ... -> While Prometheus is running, .... 
Pantheon client -> Pantheon

File: docs/Using-Pantheon/Debugging.md [hunk 7]
```diff
@@ -3,10 +3,52 @@ description: Frequently asked questions FAQ and answers for troubleshooting Pant
...
+Specify the host and port on which Prometheus accesses Pantheon client data using the CLI option
```
Comment (MadelineMurray) [inline docs/Using-Pantheon/Debugging.md]: CLI -> command line option per standards to avoid acronyms

File: docs/Using-Pantheon/Debugging.md [hunk 8]
```diff
@@ -3,10 +3,52 @@ description: Frequently asked questions FAQ and answers for troubleshooting Pant
...
+[`--metrics-listen`](../Reference/Pantheon-CLI-Syntax.md#metrics-listen). The default host and port are 127.0.0.1:9545.
+
+You can install other Prometheus components such as the Alert Manager. Once you install these components, you don't need to
+do additional configuration, because Prometheus handles and analyzes data directly from the feed.
+
+Here's an example of how you can set up and run Prometheus with Pantheon:
+
+* Install the [prometheus main component](https://prometheus.io/download/) for the Prometheus monitoring system and
+time series database on your system.
```
Comment (MadelineMurray) [inline docs/Using-Pantheon/Debugging.md]: Suggest removing 'on your system.'

File: docs/Using-Pantheon/Debugging.md [hunk 9]
```diff
@@ -3,10 +3,52 @@ description: Frequently asked questions FAQ and answers for troubleshooting Pant
...
+
+* Navigate to the directory in which you installed the Pantheon binary and run the command with the metrics options;
```
Comment (MadelineMurray) [inline docs/Using-Pantheon/Debugging.md]: If you installed the binary, you can run it from any directory.  Suggest changing to: Run Pantheon with the metrics options. For example: 
....

File: docs/Using-Pantheon/Debugging.md (hunk 8 above)
Comment (MadelineMurray) [reply docs/Using-Pantheon/Debugging.md]: You can use `brew install prometheus` on a Mac to install

File: docs/Using-Pantheon/Debugging.md [hunk 10]
```diff
@@ -3,10 +3,52 @@ description: Frequently asked questions FAQ and answers for troubleshooting Pant
...
+    ```
+
+* In another terminal, run Prometheus.
+
+* Open a web browser to `http://localhost:9090` to view the Prometheus graphical interface.
+
+* Choose **Graph** from the menu bar, then click the **Console** tab below.
+
+* From the **Insert metric at cursor** drop-down, select a metric such as `pantheon_blockchain_difficulty_total` or
+`pantheon_blockchain_height`, then click **Execute**. The values are displayed below.
```
Comment (MadelineMurray) [inline docs/Using-Pantheon/Debugging.md]: I've got Prometheus running and have added the section per Danno above to the config file but I can't see any `panetheon_*` metrics?

File: docs/Reference/Pantheon-CLI-Syntax.md (hunk 2 above)
Comment (shemnon) [reply docs/Reference/Pantheon-CLI-Syntax.md]: No, that's not what --host-whitelist is for.  It's not the best name for the flag given it's unclear what the host it is referring to.  It's the name of the host the HTTP service is serving on, so it would be 'localhost' in most cases.  It is not a whitelist of what outside clients can connect. See https://en.wikipedia.org/wiki/DNS_rebinding for what it is designed to mitigate.

File: docs/Using-Pantheon/Debugging.md (hunk 10 above)
Comment (shemnon) [reply docs/Using-Pantheon/Debugging.md]: What is the status of your 'targets'?  http://localhost:9090/targets - you should see all the targets it is trying to read.  And you can click on the endpoint for debugging as well.

Comment (MadelineMurray) [reply docs/Using-Pantheon/Debugging.md]: The prometheus one is up

File: docs/Reference/Pantheon-CLI-Syntax.md (hunk 3 above)
Comment (MadelineMurray) [reply docs/Reference/Pantheon-CLI-Syntax.md]: Done

File: docs/Using-Pantheon/Debugging.md (hunk 4 above)
Comment (MadelineMurray) [reply docs/Using-Pantheon/Debugging.md]: Done

File: docs/Using-Pantheon/Debugging.md (hunk 5 above)
Comment (MadelineMurray) [reply docs/Using-Pantheon/Debugging.md]: Done

File: docs/Using-Pantheon/Debugging.md (hunk 6 above)
Comment (MadelineMurray) [reply docs/Using-Pantheon/Debugging.md]: Done

File: docs/Using-Pantheon/Debugging.md (hunk 7 above)
Comment (MadelineMurray) [reply docs/Using-Pantheon/Debugging.md]: Done

File: docs/Using-Pantheon/Debugging.md (hunk 8 above)
Comment (MadelineMurray) [reply docs/Using-Pantheon/Debugging.md]: Done

File: docs/Using-Pantheon/Debugging.md (hunk 9 above)
Comment (MadelineMurray) [reply docs/Using-Pantheon/Debugging.md]: Done

File: docs/Using-Pantheon/Debugging.md (hunk 10 above)
Comment (MadelineMurray) [reply docs/Using-Pantheon/Debugging.md]: Changing 127.0.0.1:9545 to localhost in the prometheus.yml file or host whitelisting 127.0.01 solved this. Updating doc accordingly.

File: docs/Using-Pantheon/Debugging.md (hunk 1 above)
Comment (MadelineMurray) [reply docs/Using-Pantheon/Debugging.md]: Added - used localhost:9545 rather than 127.0.01 per comments below.
